pkg/rpcclient/actor/waiter.go

```diff
@@ -34,37 +45,60 @@ type (
 	// RPCWSWaiter is an interface that enables improved transaction awaiting functionality
 	// for Actor instance based on web-socket Block and ApplicationLog notifications.
 	RPCWSWaiter interface {
-		RPCPollingWaiter
-
+		// Context should return the RPC client context to be able to gracefully
+		// shut down all running processes (if so).
+		Context() context.Context
 		SubscribeForNewBlocksWithChan(primary *int, since *uint32, rcvrCh chan<- rpcclient.Notification) (string, error)
 		SubscribeForTransactionExecutionsWithChan(state *string, container *util.Uint256, rcvrCh chan<- rpcclient.Notification) (string, error)
 		Unsubscribe(id string) error
 	}
 )
 
-// Wait allows to wait until transaction will be accepted to the chain. It can be
-// used as a wrapper for Send or SignAndSend and accepts transaction hash,
-// ValidUntilBlock value and an error. It returns transaction execution result
-// or an error if transaction wasn't accepted to the chain.
-func (a *Actor) Wait(h util.Uint256, vub uint32, err error) (*state.AppExecResult, error) {
-	if err != nil {
-		return nil, err
+// pollingWaiter is a polling-based Waiter.
+type pollingWaiter struct {
+	RPCPollingWaiter
+	version *result.Version
+}
+
+// wsWaiter is a websocket-based Waiter.
+type wsWaiter struct {
```

Comment:
Why don't we expose these?
```
actor.PollingWaiter
actor.EventWaiter
actor.NullWaiter
```
Each will have a `New...Waiter` with an appropriate `RPC*Waiter` input (including `GetVersion` call if needed) and then you'll select an appropriate one right in the `actor.New` and that's going to be a `Waiter` implementation it'll use. It at the same time would allow to create a waiter instance without Actor, maybe not very useful, but still a possibility.

Reply:
Fixed.